lib/TwoFactorAuth.php

```diff
@@ -29,11 +28,11 @@ class TwoFactorAuth
     private static array $_base32lookup = array();
 
     public function __construct(
+        private IQRCodeProvider    $qrcodeprovider,
```

Comment:
I think this should be left where it was and you should document using named arguments in the constructor, i.e.

```php
new TwoFactorAuth(
    qrcodeprovider: $yourProvider
);
```

Reply:
oh interesting

Reply:
actually, what if people don't want a QR code out of the library? they've never known it was there and now they're forced to specify a provider they won't use?